lib/yaml/sfYamlParser.class.php

```diff
@@ -421,7 +421,7 @@ protected function parseValue($value)
     {
       $modifiers = isset($matches['modifiers']) ? $matches['modifiers'] : '';
 
-      return $this->parseFoldedScalar($matches['separator'], preg_replace('#\d+#', '', $modifiers), (int) abs($modifiers));
+      return $this->parseFoldedScalar($matches['separator'], preg_replace('#\d+#', '', $modifiers), (int) abs((int) $modifiers));
```

Comment:
```suggestion
      return $this->parseFoldedScalar($matches['separator'], preg_replace('#\d+#', '', $modifiers), abs((int) $modifiers));
```

According to the [docs](https://www.php.net/abs), `abs()` always returns in if the parameter is int which now always is, so we no longer need the previous casting.

Reply:
You're right, just tested from 5.3 to 8.1 , constat behavior, I fix it.